articles/clustering-in-unsupervised-ml/index.md

```diff
@@ -0,0 +1,157 @@
+
+###### Algorithm steps
+1.	In the first step, a core point should be identified. The core point radius is given as ε. Create a group  for each core point.
+2.	Identify border points and assign them to their designated core points. 
+3.	Any other point that is not within the group of border points or core points is treated as a nose point. This is an outlier.
+
+![DBSCAN Clustering](/engineering-education/clustering-in-unsupervised-machine-learning/dbscan-clustering.png)
+[Image Source: Git Book](https://yongle.gitbooks.io/datamining/content/figures/dbscan.png)
+
+###### Advantages 
++ It does not require the specification of the number of clusters.
++ It is very resourceful in the identification of outliers.
++ It offers flexibility in terms of the size and shape of clusters.
+
+###### Disadvantages
++ It is not effective in clustering datasets that consist of varying densities.
++ Failure to understand the data well may lead to difficulties in choosing a threshold core point radius. 
++ In some rare cases, a border point can be reached by two clusters, which may create difficulties in determining the exact cluster for the border point. 
+
+##### Gaussian Mixture Models (GMM)
+This is an advanced clustering technique in which a mixture of Gaussian distributions is used to model a dataset. These mixture models are probabilistic. GMM clustering models are used to generate data samples. 
+
+In these models, each data point is a member of all clusters in the dataset, but with varying degrees of membership. The probability of being a member of a specific cluster is between 0 and 1.
+
+In Gaussian mixture models, the main information includes the latent Gaussian centers and the covariance of data. This makes it similar to K-means clustering. 
+
+The following diagram shows a graphical representation of these models.
+![Gaussian Mixture Model](/engineering-education/clustering-in-unsupervised-machine-learning/gaussian-mixture-model.png)
+
+[Image Source: Daum](https://t1.daumcdn.net/cfile/tistory/99BC00385AC75F9A03)
+
+###### Algorithm steps
+1.	Initiate K number of Gaussian distributions. This is done using the values of standard deviation and mean. 
+2.	Expectation Phase-Assign data points to all clusters with specific membership levels.
+3.	Maximization Phase- The Gaussian parameters (mean and standard deviation) should be re-calculated using the ‘expectations’.
+4.	Evaluate whether there is convergence by examining the log-likelihood of existing data. 
+5.	Repeat steps 2-4 until there is convergence. 
+
+###### Advantages
++ It offers flexibility in terms of size and shape of clusters.
++ Membership can be assigned to multiple clusters, which makes it a fast algorithm for mixture models. 
+
+###### Disadvantages
++ If a mixture consists of insufficient points, the algorithm may diverge and establish solutions that contain infinite likelihood. This may require artificial regularization of covariance. 

```

Comment:
Artificial regularization of covariance? What does this mean, and what is its significance?

Reply:
I meant rectifying the covariance between the points artificially. I have made changes to it.